homeassistant/components/template/select.py

```diff
@@ -0,0 +1,198 @@
+"""Support for selects which integrates with other components."""
+from __future__ import annotations
+
+import contextlib
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.components.select.const import (
+    ATTR_OPTION,
+    ATTR_OPTIONS,
+    DOMAIN as SELECT_DOMAIN,
+)
+from homeassistant.components.template.trigger_entity import TriggerEntity
+from homeassistant.const import CONF_NAME, CONF_OPTIMISTIC, CONF_STATE, CONF_UNIQUE_ID
+from homeassistant.core import Config, HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.script import Script
+from homeassistant.helpers.template import Template, TemplateError
+
+from . import TriggerUpdateCoordinator
+from .const import CONF_ATTRIBUTES, CONF_AVAILABILITY
+from .template_entity import TemplateEntity
+
+CONF_SELECT_OPTION = "select_option"
+
+DEFAULT_NAME = "Template Select"
+DEFAULT_OPTIMISTIC = False
+
+SELECT_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.template,
+        vol.Required(CONF_STATE): cv.template,
+        vol.Required(CONF_SELECT_OPTION): cv.SCRIPT_SCHEMA,
+        vol.Required(CONF_ATTRIBUTES): vol.Schema(
+            {
+                vol.Required(ATTR_OPTIONS): cv.template,
+            }
+        ),
```

Comment:
Why is this an attribute? I would have expected this one to be part of the main schema and allow the attribute to be "free" list of key/value pairs defined by the user.

Reply:
do we want to support `free` lists of key/value pairs for attributes on platforms that aren't sensors? This is easy enough to change, I was just trying to match the sensor schema while acknowledging that these platforms have specific attributes that are exposed

Reply:
> do we want to support free lists of key/value pairs for attributes on platforms that aren't sensors?

I would say: not now; maybe later, maybe never.

> I was just trying to match the sensor schema while acknowledging that these platforms have specific attributes that are exposed

The sensor schema exposes the opportunity to expose custom attributes. I think this `options` (and things like `step` for number) are technically attributes but that is more a internal thing I would say.

Looking at the [Input Select](https://www.home-assistant.io/integrations/input_select), it is also a normal level parameter for `options`.